functions.js

```diff
@@ -74,11 +70,32 @@ function clickLogic (clickedCard, deck) {
             divId : clickedCard.id,
             url : clickedCard.childNodes[0].src
         }
+        turnCounter++
+        document.getElementById('turnContainer').textContent = String(turnCounter)
         if (comparison(clickedCard1.url, clickedCard2.url)) {
             clickCounter = 0
+            if(winState()) {
+                gameEnd()
+            }
         } else {
             clickCounter = 2
             cardHide(clickedCard1, clickedCard2)
         }
     }
+}
+
+function gameEnd () {
+    document.getElementById('totalTurns').textContent = String(turnCounter)
```

Comment:
also, DOM selectors are slow. reduce the amount of them you use by storing them in a variable.
`document.getElementById('totalTurns')` has been selected on line 74 already, why not store the selector in a variable globally then use it in both places.

Reply:
we have not implemented this as total turns has only been grabbed once in the DOM so a global var is not needed, line 74 is grabbing a different DOM element.